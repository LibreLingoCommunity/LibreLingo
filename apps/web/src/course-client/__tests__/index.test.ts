import { get_course, get_skill_data, get_skill_introduction } from "../index"

describe("get_course", () => {
  it("returns correct course data", async () => {
    expect(await get_course({ courseName: "test-1" })).toMatchSnapshot(
      "test course data"
    )
  })

  it("returns error when non-existent github gist is used", async () => {
    expect(
      get_course({
        courseName: "test-1",
        gistId:
          "db0545fc1ace67dd8c67d3bcae571b4442161060cd3cfc3890c55e351ec79245",
      })
    ).rejects.toThrowErrorMatchingInlineSnapshot(
      `"Could not load gist with Id "db0545fc1ace67dd8c67d3bcae571b4442161060cd3cfc3890c55e351ec79245". TypeError: Cannot convert undefined or null to object"`
    )
  })

  it("returns correct data when existing gist id is supplied", async () => {
    const courseData = await get_course({
      courseName: "test-1",
      gistId: "2428349a05d81f96b2311c2749ea5c6d",
    })

    expect(courseData).toEqual(await get_course({ courseName: "test-1" }))
  })
})

describe("get_skill_data", () => {
  it("returns correct course data", async () => {
    expect(
      await get_skill_data({ courseName: "test-1", skillName: "animals" })
    ).toMatchSnapshot()
  })

  it("returns error when non-existent github gist is used", async () => {
    expect(
      get_skill_data({
        courseName: "test-1",
        skillName: "animals",
        gistId:
          "db0545fc1ace67dd8c67d3bcae571b4442161060cd3cfc3890c55e351ec79245",
      })
    ).rejects.toThrowErrorMatchingInlineSnapshot(
      `"Could not load gist with Id "db0545fc1ace67dd8c67d3bcae571b4442161060cd3cfc3890c55e351ec79245". TypeError: Cannot convert undefined or null to object"`
    )
  })

  it("returns correct course data when gist id is supplied", async () => {
    expect(
      await get_skill_data({
        courseName: "testGist",
        skillName: "animals",
        gistId: "2428349a05d81f96b2311c2749ea5c6d",
      })
    ).toEqual({
      ...(await get_skill_data({
        courseName: "test-1",
        skillName: "animals",
      })),
      courseURL: "/course/testGist",
    })
  })
})

describe("get_skill_introduction", () => {
  it("returns correct course data", async () => {
    expect(
      await get_skill_introduction({
        courseName: "test-1",
        skillName: "animals",
      })
    ).toMatchInlineSnapshot(`
      {
        "courseName": "test-1",
        "practiceHref": "animals",
        "readmeHTML": undefined,
        "skillName": "animals",
        "title": "Animals",
      }
    `)
  })

  it("returns correct course data", async () => {
    expect(
      await get_skill_introduction({
        courseName: "testGist",
        skillName: "animals",
        gistId: "2428349a05d81f96b2311c2749ea5c6d",
      })
    ).toEqual({
      ...(await get_skill_introduction({
        courseName: "test-1",
        skillName: "animals",
      })),
      readmeHTML: "<p>hello world</p>",
      courseName: "testGist",
    })
  })
})
