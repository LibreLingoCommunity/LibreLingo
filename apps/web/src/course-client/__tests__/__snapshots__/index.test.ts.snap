// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get_course returns correct course data: test course data 1`] = `
Object {
  "courseName": "test",
  "languageCode": "test",
  "languageName": "Test Language",
  "modules": Array [
    Object {
      "skills": Array [
        Object {
          "id": "e70976d68b28",
          "levels": 1,
          "practiceHref": "short-input-test-0",
          "summary": Array [
            "perro",
            "perro",
          ],
          "title": "Short Input Test 0",
        },
        Object {
          "id": "38c2ea1c36d2",
          "imageSet": Array [
            "dog1",
            "lion3",
            "cat3",
          ],
          "introduction": "animals.md",
          "levels": 3,
          "practiceHref": "animals",
          "summary": Array [
            "perro",
            "gato",
            "oso",
            "león",
            "pato",
            "caballo",
            "el gato",
            "un oso",
            "un león",
            "el pato",
            "un caballo",
            "el perro",
            "Es un caballo",
            "Max es un perro",
            "Un pato y un caballo",
          ],
          "title": "Animals",
        },
        Object {
          "id": "a3eb592daaf6",
          "levels": 1,
          "practiceHref": "short-input-test-1",
          "summary": Array [
            "foo",
          ],
          "title": "Short Input Test 1",
        },
        Object {
          "id": "30f35fdbc4c9",
          "levels": 1,
          "practiceHref": "short-input-test-2",
          "summary": Array [
            "agua",
            "agua2",
          ],
          "title": "Short Input Test 2",
        },
        Object {
          "id": "4fdcd0f561a5",
          "levels": 1,
          "practiceHref": "short-input-test-3",
          "summary": Array [
            "Cómo estás tu?",
          ],
          "title": "Short Input Test 3",
        },
        Object {
          "id": "7d7ba7d56abb",
          "levels": 1,
          "practiceHref": "listening-test-0",
          "summary": Array [
            "perro",
            "perro",
          ],
          "title": "Listening test 0",
        },
        Object {
          "id": "07b599369280",
          "levels": 2,
          "practiceHref": "options-test-0",
          "summary": Array [
            "perro",
            "león",
            "cerdo",
          ],
          "title": "Options test 0",
        },
        Object {
          "id": "6ef61f72a217",
          "levels": 1,
          "practiceHref": "chips-test-0",
          "summary": Array [
            "¿Como, estás hoy?",
            "Tu como estás hoy?",
          ],
          "title": "Chips test 0",
        },
        Object {
          "id": "42f05dd67031",
          "levels": 1,
          "practiceHref": "continuous",
          "summary": Array [],
          "title": "Continuous",
        },
        Object {
          "id": "a8dbe791bff7",
          "levels": 2,
          "practiceHref": "cards-test",
          "summary": Array [
            "pasta",
            "pasta1",
            "pasta2",
            "pasta3",
          ],
          "title": "Cards test",
        },
      ],
      "title": "Basics",
    },
  ],
  "repositoryURL": "https://github.com/LibreLingo/LibreLingo/",
  "specialCharacters": Array [
    "á",
    "Á",
    "é",
    "É",
    "í",
    "Í",
    "ó",
    "Ó",
    "ú",
    "Ú",
    "ü",
    "Ü",
    "ñ",
    "Ñ",
    "¿",
    "¡",
  ],
}
`;

exports[`get_skill_data returns correct course data 1`] = `
Object {
  "challengesPerLevel": 16,
  "courseURL": "/course/test",
  "languageCode": "test",
  "languageName": "Test Language",
  "rawChallenges": Array [
    Object {
      "formInTargetLanguage": "Es un caballo",
      "group": "353b3be9ecc5",
      "id": "d8fba73a8b08",
      "meaningInSourceLanguage": "It's a horse",
      "priority": 0,
      "type": "options",
    },
    Object {
      "answer": "Es un caballo",
      "audio": "c8caa295c49f0dcf7cdeb74f620d8df89c0c15f28aff535d863959e47657ae5f",
      "group": "353b3be9ecc5",
      "id": "a72df7965e1a",
      "meaning": "It's a horse",
      "priority": 1,
      "type": "listeningExercise",
    },
    Object {
      "chips": Array [
        "Es",
        "un",
        "caballo",
        "es",
        "can",
      ],
      "formattedSolution": "Es un caballo",
      "group": "353b3be9ecc5",
      "id": "390f4df6b328",
      "phrase": Array [
        Object {
          "definition": "es",
          "word": "It's",
        },
        Object {
          "definition": "un
una",
          "word": "a",
        },
        Object {
          "definition": "caballo",
          "word": "horse",
        },
      ],
      "priority": 2,
      "solutions": Array [
        Array [
          "Es",
          "un",
          "caballo",
        ],
      ],
      "translatesToSourceLanguage": false,
      "type": "chips",
    },
    Object {
      "chips": Array [
        "It's",
        "a",
        "horse",
        "is",
        "are",
      ],
      "formattedSolution": "It's a horse",
      "group": "353b3be9ecc5",
      "id": "b1e719e8de24",
      "phrase": Array [
        Object {
          "definition": "is",
          "word": "Es",
        },
        Object {
          "definition": "a/an (masc.)",
          "word": "un",
        },
        Object {
          "definition": "horse",
          "word": "caballo",
        },
      ],
      "priority": 2,
      "solutions": Array [
        Array [
          "It's",
          "a",
          "horse",
        ],
      ],
      "translatesToSourceLanguage": true,
      "type": "chips",
    },
    Object {
      "formInTargetLanguage": "Max es un perro",
      "group": "3b3665db19af",
      "id": "b1ae0631e52a",
      "meaningInSourceLanguage": "Max is a dog",
      "priority": 0,
      "type": "options",
    },
    Object {
      "answer": "Max es un perro",
      "audio": "d4d634de9930f4a01fab5acb342abcaa492d1558d3788956c9ff4e12f88f5dc8",
      "group": "3b3665db19af",
      "id": "a7869436300f",
      "meaning": "Max is a dog",
      "priority": 1,
      "type": "listeningExercise",
    },
    Object {
      "chips": Array [
        "Max",
        "es",
        "un",
        "perro",
        "Es",
        "Un",
        "can",
      ],
      "formattedSolution": "Max es un perro",
      "group": "3b3665db19af",
      "id": "a2edb67c79cf",
      "phrase": Array [
        Object {
          "definition": "Max (name)",
          "word": "Max",
        },
        Object {
          "definition": "es",
          "word": "is",
        },
        Object {
          "definition": "un
una",
          "word": "a",
        },
        Object {
          "definition": "can
perro",
          "word": "dog",
        },
      ],
      "priority": 2,
      "solutions": Array [
        Array [
          "Max",
          "es",
          "un",
          "perro",
        ],
        Array [
          "Max",
          "es",
          "un",
          "can",
        ],
      ],
      "translatesToSourceLanguage": false,
      "type": "chips",
    },
    Object {
      "chips": Array [
        "Max",
        "is",
        "a",
        "dog",
        "A",
        "pig",
        "and",
      ],
      "formattedSolution": "Max is a dog",
      "group": "3b3665db19af",
      "id": "0de5f458a546",
      "phrase": Array [
        Object {
          "definition": "Max (name)",
          "word": "Max",
        },
        Object {
          "definition": "is",
          "word": "es",
        },
        Object {
          "definition": "a/an (masc.)",
          "word": "un",
        },
        Object {
          "definition": "dog
the dog",
          "word": "perro",
        },
      ],
      "priority": 2,
      "solutions": Array [
        Array [
          "Max",
          "is",
          "a",
          "dog",
        ],
      ],
      "translatesToSourceLanguage": true,
      "type": "chips",
    },
    Object {
      "formInTargetLanguage": "Un pato y un caballo",
      "group": "882c91cd979e",
      "id": "a6bc83a45629",
      "meaningInSourceLanguage": "A duck and a horse",
      "priority": 0,
      "type": "options",
    },
    Object {
      "answer": "Un pato y un caballo",
      "audio": "02ef373b1ecd6cb272d135a9d9401f83038eec1a083eaae99adecd0cc8822d22",
      "group": "882c91cd979e",
      "id": "93c5603e2bed",
      "meaning": "A duck and a horse",
      "priority": 1,
      "type": "listeningExercise",
    },
    Object {
      "chips": Array [
        "Un",
        "pato",
        "y",
        "un",
        "caballo",
        "can",
        "Es",
        "Tu",
        "Max",
      ],
      "formattedSolution": "Un pato y un caballo",
      "group": "882c91cd979e",
      "id": "07d42f76c5cf",
      "phrase": Array [
        Object {
          "definition": "un
una",
          "word": "A",
        },
        Object {
          "definition": "pato",
          "word": "duck",
        },
        Object {
          "definition": "y",
          "word": "and",
        },
        Object {
          "definition": "un
una",
          "word": "a",
        },
        Object {
          "definition": "caballo",
          "word": "horse",
        },
      ],
      "priority": 2,
      "solutions": Array [
        Array [
          "Un",
          "pato",
          "y",
          "un",
          "caballo",
        ],
      ],
      "translatesToSourceLanguage": false,
      "type": "chips",
    },
    Object {
      "chips": Array [
        "A",
        "duck",
        "and",
        "a",
        "horse",
        "are",
        "is",
        "dog",
        "Max",
      ],
      "formattedSolution": "A duck and a horse",
      "group": "882c91cd979e",
      "id": "d6182c2ffeec",
      "phrase": Array [
        Object {
          "definition": "a/an (masc.)",
          "word": "Un",
        },
        Object {
          "definition": "duck",
          "word": "pato",
        },
        Object {
          "definition": "and",
          "word": "y",
        },
        Object {
          "definition": "a/an (masc.)",
          "word": "un",
        },
        Object {
          "definition": "horse",
          "word": "caballo",
        },
      ],
      "priority": 2,
      "solutions": Array [
        Array [
          "A",
          "duck",
          "and",
          "a",
          "horse",
        ],
      ],
      "translatesToSourceLanguage": true,
      "type": "chips",
    },
    Object {
      "formInTargetLanguage": "perro",
      "group": "80ce380e3103",
      "id": "34538f2dd5fb",
      "meaningInSourceLanguage": "dog",
      "pictures": Array [
        "dog1.jpg",
        "dog2.jpg",
        "dog3.jpg",
      ],
      "priority": 0,
      "type": "cards",
    },
    Object {
      "formInTargetLanguage": Array [
        "perro",
        "can",
      ],
      "group": "80ce380e3103",
      "id": "5525da287ae4",
      "phrase": Array [
        Object {
          "definition": "can
perro",
          "word": "dog",
        },
      ],
      "pictures": Array [
        "dog1.jpg",
        "dog2.jpg",
        "dog3.jpg",
      ],
      "priority": 1,
      "type": "shortInput",
    },
    Object {
      "answer": "perro",
      "audio": "ec7c10ee24ad5f0e4ce849100627093c241de12de6f58fe86eccd0e15a86d98e",
      "group": "80ce380e3103",
      "id": "b8cf67f2cddb",
      "meaning": "dog",
      "priority": 1,
      "type": "listeningExercise",
    },
    Object {
      "formInTargetLanguage": "gato",
      "group": "499c2ff87207",
      "id": "8b5c69a1d503",
      "meaningInSourceLanguage": "cat",
      "pictures": Array [
        "cat1.jpg",
        "cat2.jpg",
        "cat3.jpg",
      ],
      "priority": 0,
      "type": "cards",
    },
    Object {
      "formInTargetLanguage": Array [
        "gato",
      ],
      "group": "499c2ff87207",
      "id": "10a8fbd5bdb6",
      "phrase": Array [
        Object {
          "definition": "gato",
          "word": "cat",
        },
      ],
      "pictures": Array [
        "cat1.jpg",
        "cat2.jpg",
        "cat3.jpg",
      ],
      "priority": 1,
      "type": "shortInput",
    },
    Object {
      "answer": "gato",
      "audio": "82c40010d4dab18b64ee7dac690afa6b2d0841b485d1205421c8cf0eb99177d1",
      "group": "499c2ff87207",
      "id": "7378cf4446f5",
      "meaning": "cat",
      "priority": 1,
      "type": "listeningExercise",
    },
    Object {
      "formInTargetLanguage": "oso",
      "group": "5fe33898f3f4",
      "id": "a29a33e3d267",
      "meaningInSourceLanguage": "bear",
      "pictures": Array [
        "bear1.jpg",
        "bear2.jpg",
        "bear1.jpg",
      ],
      "priority": 0,
      "type": "cards",
    },
    Object {
      "formInTargetLanguage": Array [
        "oso",
      ],
      "group": "5fe33898f3f4",
      "id": "cacb8f13a3e8",
      "phrase": Array [
        Object {
          "definition": "oso",
          "word": "bear",
        },
      ],
      "pictures": Array [
        "bear1.jpg",
        "bear2.jpg",
        "bear1.jpg",
      ],
      "priority": 1,
      "type": "shortInput",
    },
    Object {
      "answer": "oso",
      "audio": "c33fb805b0edcaa8c64980e5055be2a3e739bdb56a185f0a00eea0bec361c6a7",
      "group": "5fe33898f3f4",
      "id": "a3b5da93dda3",
      "meaning": "bear",
      "priority": 1,
      "type": "listeningExercise",
    },
    Object {
      "formInTargetLanguage": "león",
      "group": "f1c4f9069794",
      "id": "8a293244390a",
      "meaningInSourceLanguage": "lion",
      "pictures": Array [
        "lion1.jpg",
        "lion2.jpg",
        "lion3.jpg",
      ],
      "priority": 0,
      "type": "cards",
    },
    Object {
      "formInTargetLanguage": Array [
        "león",
      ],
      "group": "f1c4f9069794",
      "id": "59cc971773bd",
      "phrase": Array [
        Object {
          "definition": "león",
          "word": "lion",
        },
      ],
      "pictures": Array [
        "lion1.jpg",
        "lion2.jpg",
        "lion3.jpg",
      ],
      "priority": 1,
      "type": "shortInput",
    },
    Object {
      "answer": "león",
      "audio": "61c49163c6b66e815712f76c7d29526cae2869360adb30072d50064af6b6febe",
      "group": "f1c4f9069794",
      "id": "4f94d302e9f1",
      "meaning": "lion",
      "priority": 1,
      "type": "listeningExercise",
    },
    Object {
      "formInTargetLanguage": "pato",
      "group": "42b787861888",
      "id": "657d03aa4080",
      "meaningInSourceLanguage": "duck",
      "pictures": Array [
        "duck1.jpg",
        "duck2.jpg",
        "duck3.jpg",
      ],
      "priority": 0,
      "type": "cards",
    },
    Object {
      "formInTargetLanguage": Array [
        "pato",
      ],
      "group": "42b787861888",
      "id": "4abdbad9d94b",
      "phrase": Array [
        Object {
          "definition": "pato",
          "word": "duck",
        },
      ],
      "pictures": Array [
        "duck1.jpg",
        "duck2.jpg",
        "duck3.jpg",
      ],
      "priority": 1,
      "type": "shortInput",
    },
    Object {
      "answer": "pato",
      "audio": "f8cdc4d959e3901e785f24031adb3278817f8a0eedab085d505f84f8202e902a",
      "group": "42b787861888",
      "id": "5a038c083296",
      "meaning": "duck",
      "priority": 1,
      "type": "listeningExercise",
    },
    Object {
      "formInTargetLanguage": "caballo",
      "group": "80881eb450de",
      "id": "281615db0129",
      "meaningInSourceLanguage": "horse",
      "pictures": Array [
        "horse1.jpg",
        "horse2.jpg",
        "horse3.jpg",
      ],
      "priority": 0,
      "type": "cards",
    },
    Object {
      "formInTargetLanguage": Array [
        "caballo",
      ],
      "group": "80881eb450de",
      "id": "8c50d4ba8e87",
      "phrase": Array [
        Object {
          "definition": "caballo",
          "word": "horse",
        },
      ],
      "pictures": Array [
        "horse1.jpg",
        "horse2.jpg",
        "horse3.jpg",
      ],
      "priority": 1,
      "type": "shortInput",
    },
    Object {
      "answer": "caballo",
      "audio": "4a118d1b5f28e5c8caba8124691d13560a68e54dad613867f174a0fccb8a0036",
      "group": "80881eb450de",
      "id": "8412170ef145",
      "meaning": "horse",
      "priority": 1,
      "type": "listeningExercise",
    },
    Object {
      "formInTargetLanguage": "el gato",
      "group": "4ee3354aceed",
      "id": "ee1532beb159",
      "meaningInSourceLanguage": "the cat",
      "pictures": Array [
        "cat1.jpg",
        "cat2.jpg",
        "cat3.jpg",
      ],
      "priority": 0,
      "type": "cards",
    },
    Object {
      "formInTargetLanguage": Array [
        "el gato",
      ],
      "group": "4ee3354aceed",
      "id": "07228bae98c2",
      "phrase": Array [
        Object {
          "definition": "el
la",
          "word": "the",
        },
        Object {
          "definition": "gato",
          "word": "cat",
        },
      ],
      "pictures": Array [
        "cat1.jpg",
        "cat2.jpg",
        "cat3.jpg",
      ],
      "priority": 1,
      "type": "shortInput",
    },
    Object {
      "answer": "el gato",
      "audio": "4bb467d83dad76412c01811549cc2338bedaefc54995c3cf67bb3e41cf19fc8a",
      "group": "4ee3354aceed",
      "id": "86b02f894a50",
      "meaning": "the cat",
      "priority": 1,
      "type": "listeningExercise",
    },
    Object {
      "formInTargetLanguage": "un oso",
      "group": "bd9d31c43fd7",
      "id": "057fb513e788",
      "meaningInSourceLanguage": "a bear",
      "pictures": Array [
        "bear1.jpg",
        "bear2.jpg",
        "bear1.jpg",
      ],
      "priority": 0,
      "type": "cards",
    },
    Object {
      "formInTargetLanguage": Array [
        "un oso",
      ],
      "group": "bd9d31c43fd7",
      "id": "6d217d2824f6",
      "phrase": Array [
        Object {
          "definition": "un
una",
          "word": "a",
        },
        Object {
          "definition": "oso",
          "word": "bear",
        },
      ],
      "pictures": Array [
        "bear1.jpg",
        "bear2.jpg",
        "bear1.jpg",
      ],
      "priority": 1,
      "type": "shortInput",
    },
    Object {
      "answer": "un oso",
      "audio": "f3df3caf8a295bce0cc892e6e4fd5857567a0ecf1f1ca6b229fa32a7932badac",
      "group": "bd9d31c43fd7",
      "id": "264445f847e0",
      "meaning": "a bear",
      "priority": 1,
      "type": "listeningExercise",
    },
    Object {
      "formInTargetLanguage": "un león",
      "group": "3958592ea672",
      "id": "e58fa078f707",
      "meaningInSourceLanguage": "a lion",
      "pictures": Array [
        "lion1.jpg",
        "lion2.jpg",
        "lion3.jpg",
      ],
      "priority": 0,
      "type": "cards",
    },
    Object {
      "formInTargetLanguage": Array [
        "un león",
      ],
      "group": "3958592ea672",
      "id": "8aa4a159aa9d",
      "phrase": Array [
        Object {
          "definition": "un
una",
          "word": "a",
        },
        Object {
          "definition": "león",
          "word": "lion",
        },
      ],
      "pictures": Array [
        "lion1.jpg",
        "lion2.jpg",
        "lion3.jpg",
      ],
      "priority": 1,
      "type": "shortInput",
    },
    Object {
      "answer": "un león",
      "audio": "35c3e015c58213f911089e32ca76728a423c0686b43b2069efc3a51eb2f0b55d",
      "group": "3958592ea672",
      "id": "e84c3a852322",
      "meaning": "a lion",
      "priority": 1,
      "type": "listeningExercise",
    },
    Object {
      "formInTargetLanguage": "el pato",
      "group": "0e72f647be1e",
      "id": "f3ec01d6882e",
      "meaningInSourceLanguage": "the duck",
      "pictures": Array [
        "duck1.jpg",
        "duck2.jpg",
        "duck3.jpg",
      ],
      "priority": 0,
      "type": "cards",
    },
    Object {
      "formInTargetLanguage": Array [
        "el pato",
      ],
      "group": "0e72f647be1e",
      "id": "06a6fc0c0805",
      "phrase": Array [
        Object {
          "definition": "el
la",
          "word": "the",
        },
        Object {
          "definition": "pato",
          "word": "duck",
        },
      ],
      "pictures": Array [
        "duck1.jpg",
        "duck2.jpg",
        "duck3.jpg",
      ],
      "priority": 1,
      "type": "shortInput",
    },
    Object {
      "answer": "el pato",
      "audio": "c9e375874e561dfc71c50ce870b0a7a96b83170a9d0fb7913b96907e69c9c43a",
      "group": "0e72f647be1e",
      "id": "b7b8177ac94a",
      "meaning": "the duck",
      "priority": 1,
      "type": "listeningExercise",
    },
    Object {
      "formInTargetLanguage": "un caballo",
      "group": "39ca9bde2b5b",
      "id": "e5b5219601d5",
      "meaningInSourceLanguage": "a horse",
      "pictures": Array [
        "horse1.jpg",
        "horse2.jpg",
        "horse3.jpg",
      ],
      "priority": 0,
      "type": "cards",
    },
    Object {
      "formInTargetLanguage": Array [
        "un caballo",
      ],
      "group": "39ca9bde2b5b",
      "id": "d1ef1f9a3927",
      "phrase": Array [
        Object {
          "definition": "un
una",
          "word": "a",
        },
        Object {
          "definition": "caballo",
          "word": "horse",
        },
      ],
      "pictures": Array [
        "horse1.jpg",
        "horse2.jpg",
        "horse3.jpg",
      ],
      "priority": 1,
      "type": "shortInput",
    },
    Object {
      "answer": "un caballo",
      "audio": "d8bc42f9d887eb6b01dbd15b8c3fb752559b6423568bed57a49d331b30147297",
      "group": "39ca9bde2b5b",
      "id": "8ea464c11625",
      "meaning": "a horse",
      "priority": 1,
      "type": "listeningExercise",
    },
    Object {
      "formInTargetLanguage": "el perro",
      "group": "dc51787e55a3",
      "id": "26eaf9940451",
      "meaningInSourceLanguage": "the dog",
      "pictures": Array [
        "dog1.jpg",
        "dog2.jpg",
        "dog3.jpg",
      ],
      "priority": 0,
      "type": "cards",
    },
    Object {
      "formInTargetLanguage": Array [
        "el perro",
        "el can",
      ],
      "group": "dc51787e55a3",
      "id": "b174b698c4cd",
      "phrase": Array [
        Object {
          "definition": "el
la",
          "word": "the",
        },
        Object {
          "definition": "can
perro",
          "word": "dog",
        },
      ],
      "pictures": Array [
        "dog1.jpg",
        "dog2.jpg",
        "dog3.jpg",
      ],
      "priority": 1,
      "type": "shortInput",
    },
    Object {
      "answer": "el perro",
      "audio": "2184ed4937d5df0e572d49d89248d5b503b3a0a4c27f03dcf09e11f3b74a010a",
      "group": "dc51787e55a3",
      "id": "c61ecac20453",
      "meaning": "the dog",
      "priority": 1,
      "type": "listeningExercise",
    },
  ],
  "repositoryURL": "https://github.com/LibreLingo/LibreLingo/",
  "skillId": "38c2ea1c36d2",
  "skillName": "animals",
  "specialCharacters": Array [
    "á",
    "Á",
    "é",
    "É",
    "í",
    "Í",
    "ó",
    "Ó",
    "ú",
    "Ú",
    "ü",
    "Ü",
    "ñ",
    "Ñ",
    "¿",
    "¡",
  ],
}
`;
